tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -109,6 +109,22 @@ def read(self, filename, binary_mode=False, size=None, offset=None):
             else:
                 return f.read()
 
+    def write(self, filename, file_content, binary_mode=False):
+        """Writes string file contents to a file.
+
+        Args:
+            filename: string, a path
+            file_content: string, the contents
+            binary_mode: bool, write as binary if True, otherwise text
+        """
+        mode = "wb" if binary_mode else "w"
+        if binary_mode:
+            file_content = compat.as_bytes(file_content)
+        else:
+            file_content = compat.as_text(file_content)
+        with io.open(filename, mode) as f:
```

Comment:
It'd be better to set encoding explicitly (e.g. to UTF-8) versus letting it be platform-dependent, for the text-mode case.